commands/ls.go

```diff
@@ -67,3 +80,48 @@ func cmdLs(c *cli.Context) {
 
 	w.Flush()
 }
+
+func parseFilters(filters []string) (options FilterOptions, err error) {
+	options = FilterOptions{}
+	for _, f := range filters {
+		kv := strings.SplitN(f, "=", 2)
+		key, value := kv[0], kv[1]
+
+		switch key {
+		case "swarm":
+			options.SwarmName = value
+		case "driver":
+			options.DriverName = value
+		default:
+			err = fmt.Errorf("Unsupported filter key '%s'", key)
+		}
+	}
+	return
```

Comment:
I was using a "naked return" (so, equivalent to `return options, err`), but I see that's not very clear. I'll make it explicit :)